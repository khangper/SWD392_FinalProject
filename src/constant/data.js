export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
