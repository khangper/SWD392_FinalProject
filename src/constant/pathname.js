export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
