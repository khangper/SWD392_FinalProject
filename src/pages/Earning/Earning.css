.earning-coin-icon {
  width: 20px;
  height: 20px;
  position: relative;
  flex-shrink: 0;
}

.earning-icon {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-end;
  padding: 0 0 2.8px;
}

.earning-earning {
  position: relative;
  line-height: 26px;
  font-weight: 550;
  color: #333;
  display: inline-block;
  min-width: 73px;
  font-size: 18px;
}

.earning-label,
.earning-title {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
}

.earning-label {
  align-items: flex-end;
  gap: 4px;
}

.earning-title {
  align-items: flex-start;
}

.earning-heading-1,
.earning-heading-2 {
  position: relative;
  line-height: 26px;
}

.earning-heading-1 {
  align-self: stretch;
  font-size: 14px;
}

.earning-heading-2 {
  font-weight: 600;
  white-space: nowrap;
  font-size: 30px;
}

.earning-heading-2-wrapper {
  align-self: stretch;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: center;
  padding: 0 0 2px;
}

.earning-heading-1-wrapper {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.earning-paragraph-background {
  border-radius: 3px;
  background-color: #333;
  color: #fff;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  gap: 20px;
  padding: 30px 10px 55px;
  min-width: 250px;
  min-height: 157px;
  max-width: 100%;
  width: 100%;
}

.earning-summary {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: space-between;
  gap: 20px;
  max-width: 100%;
  text-align: center;
  width: 100%;
}

.earning-table-countries {
  text-align: left;
  width: 100%;
  border-collapse: collapse;
  border: 1px solid #e5e5e5;
  background-color: #fff;
  border-radius: 3px;
}

.earning-table-countries td {
  padding: 15px;
  border-bottom: 1px solid #e5e5e5;
  font-size: 14px;
  color: #686f7a;
  font-weight: 550;
}
.earning-table-countries th {
  border-bottom: 1px solid #e5e5e5;
}

.earning-table-top-countries {
  padding: 15px 15px 15px 20px;
  font-size: 16px;

  color: #333;
}

.earning-table-coin {
  text-align: right;
}

.earning-item-sale select {
  width: 150px;
  max-width: 100%;
  min-width: 100%;
  padding: 10px;
  background: #fff;
  border: 1px solid #e5e5e5 !important;
  outline: 0;
  color: rgba(0, 0, 0, 0.87);
  border-radius: 3px !important;
  box-shadow: inset 0px 1px 4px rgba(0, 0, 0, 0.02) !important;
  font-size: 15px;
  cursor: pointer;
}

.earning-time-sale{
    display: flex;
    gap: 20px;
    width: 100%;
    align-items: flex-start;
    justify-content: flex-start;
}

.earning-items{
    display: flex;
    gap: 7px;
    padding-top: 10px;
    align-items: flex-start;
    justify-content: flex-start;
    font-size: 15px;
    font-family: inherit;
    font-weight: 500;
    color: #333;
}

.earning-symbol{
    color: #a1a1a1;
}

.earning-items a{
    color: #4183c4;
    cursor: pointer;
}

.earning-items a:hover{
    color: #a1a1a1;
}

.earning-table-items{
    text-align: left;
    font-size: 15px;
    border: 1px solid #e5e5e5;
    width: 100%;
    color: #333;
    border-collapse: collapse;
    border-radius: 3px;
}

.earning-table-items th, .earning-table-items td{
    padding: 15px;
    border-bottom: 1px solid #e5e5e5;
}

.earning-table-items tfoot{
    color: #fff;
    background-color: #333;
    font-weight: 550;
}

.earning-table-items thead{
    background-color: #ffecec;
}

.earning-table-items tbody{
    background-color: #fff;
}

.earning-table-wrapper{
    display: flex;
    flex-direction: row;
    flex: 1;
    gap: 20px;
    flex-wrap: wrap;
}

.earning-table-1{
    flex: 0.328;
    min-width: 330px;
}

.earning-table-2{
    flex: 0.672;
    min-width: 680px;
    display: flex;
    flex-direction: column;
    gap: 30px;
}

.earning-wrapper{
    display: flex;
    flex-direction: column;
    align-items: normal;
    justify-content: flex-start;
    gap: 50px;
    flex-wrap: wrap;
    padding-bottom: 30px;
    padding-left: 10px;
}

@media screen and (max-width:1350px){
    .earning-table-wrapper{

        display: block;

    }

    .earning-table-2{
        padding-top: 50px;
    }

}