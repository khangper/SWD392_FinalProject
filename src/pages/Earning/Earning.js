import React, { useEffect } from "react";
import "./Earning.css";
import { MdOutlineAttachMoney } from "react-icons/md";
import { useDispatch, useSelector } from "react-redux";
import { fetchEarningTableRequest } from "../../redux/reduxActions/earningActions/EarningTableAction";
import { fetchEarningTopCountryRequest } from "../../redux/reduxActions/earningActions/EarningTopCountryAction";

const Earning = () => {
  const dispatch = useDispatch();
  const { itemSales } = useSelector((state) => state.earning_table);
  const { topCountries } = useSelector((state) => state.earning_top_country);

  useEffect(() => {
    dispatch(fetchEarningTableRequest());
    dispatch(fetchEarningTopCountryRequest());
  }, [dispatch]);

  //Calculate the total earnings
  const totalEarning = itemSales.reduce((acc, itemSale) => {
    return acc + itemSale.earning;
  }, 0);

  const formattedTotalEarning = totalEarning.toLocaleString('en-US', {
  style: 'currency',
  currency: 'USD',
  minimumFractionDigits: 2,
  maximumFractionDigits: 2
});

  //Calculate thr total item
  const totalItem = itemSales.reduce((acc, itemSale) => {
    return acc + itemSale.numberOfItems;
  }, 0);

  return (
    <main className="earning">
      <div className="earning-wrapper">
        <div className="earning-title">
          <div className="earning-label">
            <div className="earning-icon">
              <MdOutlineAttachMoney className="earning-coin-icon" />
            </div>
            <h3 className="earning-earning">Earning</h3>
          </div>
        </div>

        <div className="earning-summary">
          <div className="earning-paragraph-background">
            <div className="earning-heading-1-wrapper">
              <div className="earning-heading-1">
                Sales earnings this month (April), after edututs+ fees, & before
                taxes:
              </div>
            </div>
            <div className="earning-heading-2-wrapper">
              <div className="earning-heading-2">$1,146.78</div>
            </div>
          </div>
          <div className="earning-paragraph-background">
            <div className="earning-heading-1-wrapper">
              <div className="earning-heading-1">Your balance:</div>
            </div>
            <div className="earning-heading-2-wrapper">
              <div className="earning-heading-2">$1,146.78</div>
            </div>
          </div>
          <div className="earning-paragraph-background">
            <div className="earning-heading-1-wrapper">
              <div className="earning-heading-1">
                Total value of your sales, before taxes:
              </div>
            </div>
            <div className="earning-heading-2-wrapper">
              <div className="earning-heading-2">$95,895.54</div>
            </div>
          </div>
        </div>

        <div className="earning-table-wrapper">
          <div className="earning-table-1">
            <table className="earning-table-countries">
              <thead>
                <tr>
                  <th className="earning-table-top-countries">
                    Your Top Countries
                  </th>
                  <th></th>
                </tr>
              </thead>
              <tbody>
                {topCountries.map((country) => (
                  <tr key={country.id}>
                    <td>{country.country}</td>
                    <td className="earning-table-coin">${country.price}</td>
                  </tr>
                ))}
              </tbody>
            </table>
          </div>

          <div className="earning-table-2">
            <div className="earning-time-sale">
              <div className="earning-item-sale">
                <select name="sale" id="sale">
                  <option value="" hidden>
                    Item Sales
                  </option>
                  <option value="total">Total Sales</option>
                  <option value="2020">2020</option>
                </select>
              </div>
              <div className="earning-items">
                <a href="#"> All Time</a>
                <div className="earning-symbol">/</div>
                <a href="#">2020</a>
                <div className="earning-symbol">/</div>
                <div className="earning-month">April</div>
              </div>
            </div>

            <div className="earning-table-data">
              <table className="earning-table-items">
                <thead>
                  <tr>
                    <th>Date</th>
                    <th>Item Sales Counts</th>
                    <th>Earning</th>
                  </tr>
                </thead>
                <tbody>
                  {itemSales.map((item) => (
                    <tr key={item.id}>
                      <td>{item.date}</td>
                      <td>{item.numberOfItems}</td>
                      <td>{item.formattedEarning}</td>
                    </tr>
                  ))}
                </tbody>
                <tfoot>
                  <tr>
                    <td>Total</td>
                    <td>{totalItem}</td>
                    <td>{formattedTotalEarning}</td>
                  </tr>
                </tfoot>
              </table>
            </div>
          </div>
        </div>
      </div>
    </main>
  );
};

export default Earning;
