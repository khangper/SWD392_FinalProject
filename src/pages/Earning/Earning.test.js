import React from "react";
import { render, screen } from "@testing-library/react";
import Earning from "../Earning/Earning";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../constant/test-util";

describe("Earning", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the Earning component", async () => {
    const { container } = renderWithProviders(<Earning />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
