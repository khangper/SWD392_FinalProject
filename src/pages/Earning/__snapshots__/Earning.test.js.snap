// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Earning renders the Earning component 1`] = `
<div>
  <main
    class="earning"
  >
    <div
      class="earning-wrapper"
    >
      <div
        class="earning-title"
      >
        <div
          class="earning-label"
        >
          <div
            class="earning-icon"
          >
            <svg
              class="earning-coin-icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M11.8 10.9c-2.27-.59-3-1.2-3-2.15 0-1.09 1.01-1.85 2.7-1.85 1.78 0 2.44.85 2.5 2.1h2.21c-.07-1.72-1.12-3.3-3.21-3.81V3h-3v2.16c-1.94.42-3.5 1.68-3.5 3.61 0 2.31 1.91 3.46 4.7 4.13 2.5.6 3 1.48 3 2.41 0 .69-.49 1.79-2.7 1.79-2.06 0-2.87-.92-2.98-2.1h-2.2c.12 2.19 1.76 3.42 3.68 3.83V21h3v-2.15c1.95-.37 3.5-1.5 3.5-3.55 0-2.84-2.43-3.81-4.7-4.4z"
              />
            </svg>
          </div>
          <h3
            class="earning-earning"
          >
            Earning
          </h3>
        </div>
      </div>
      <div
        class="earning-summary"
      >
        <div
          class="earning-paragraph-background"
        >
          <div
            class="earning-heading-1-wrapper"
          >
            <div
              class="earning-heading-1"
            >
              Sales earnings this month (April), after edututs+ fees, & before taxes:
            </div>
          </div>
          <div
            class="earning-heading-2-wrapper"
          >
            <div
              class="earning-heading-2"
            >
              $1,146.78
            </div>
          </div>
        </div>
        <div
          class="earning-paragraph-background"
        >
          <div
            class="earning-heading-1-wrapper"
          >
            <div
              class="earning-heading-1"
            >
              Your balance:
            </div>
          </div>
          <div
            class="earning-heading-2-wrapper"
          >
            <div
              class="earning-heading-2"
            >
              $1,146.78
            </div>
          </div>
        </div>
        <div
          class="earning-paragraph-background"
        >
          <div
            class="earning-heading-1-wrapper"
          >
            <div
              class="earning-heading-1"
            >
              Total value of your sales, before taxes:
            </div>
          </div>
          <div
            class="earning-heading-2-wrapper"
          >
            <div
              class="earning-heading-2"
            >
              $95,895.54
            </div>
          </div>
        </div>
      </div>
      <div
        class="earning-table-wrapper"
      >
        <div
          class="earning-table-1"
        >
          <table
            class="earning-table-countries"
          >
            <thead>
              <tr>
                <th
                  class="earning-table-top-countries"
                >
                  Your Top Countries
                </th>
                <th />
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
        <div
          class="earning-table-2"
        >
          <div
            class="earning-time-sale"
          >
            <div
              class="earning-item-sale"
            >
              <select
                id="sale"
                name="sale"
              >
                <option
                  hidden=""
                  value=""
                >
                  Item Sales
                </option>
                <option
                  value="total"
                >
                  Total Sales
                </option>
                <option
                  value="2020"
                >
                  2020
                </option>
              </select>
            </div>
            <div
              class="earning-items"
            >
              <a
                href="#"
              >
                 All Time
              </a>
              <div
                class="earning-symbol"
              >
                /
              </div>
              <a
                href="#"
              >
                2020
              </a>
              <div
                class="earning-symbol"
              >
                /
              </div>
              <div
                class="earning-month"
              >
                April
              </div>
            </div>
          </div>
          <div
            class="earning-table-data"
          >
            <table
              class="earning-table-items"
            >
              <thead>
                <tr>
                  <th>
                    Date
                  </th>
                  <th>
                    Item Sales Counts
                  </th>
                  <th>
                    Earning
                  </th>
                </tr>
              </thead>
              <tbody />
              <tfoot>
                <tr>
                  <td>
                    Total
                  </td>
                  <td>
                    0
                  </td>
                  <td>
                    $0.00
                  </td>
                </tr>
              </tfoot>
            </table>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
